examples/step-55/step-55.cc

```diff
@@ -193,18 +193,20 @@ namespace Step55
 
     virtual void vector_value(const Point<dim> &p,
                               Vector<double> &  value) const override;
+
+  private:
+    static constexpr double pi  = numbers::PI;
+    static constexpr double pi2 = numbers::PI * numbers::PI;
   };
 
 
   template <int dim>
   void RightHandSide<dim>::vector_value(const Point<dim> &p,
                                         Vector<double> &  values) const
   {
-    const double R_x = p[0];
-    const double R_y = p[1];
+    const double &R_x = p[0];
+    const double &R_y = p[1];
```

Comment:
You make it more difficult for the compiler to reason about whether `R_x` could change from one call to another. For example, in the expression below, the compiler will not be able to tell whether `R_x` could change in the calls to `std::sqrt` because it could point to a global variable that `std::sqrt` clobbers. If you leave it as a local copy, the compiler will know that that can't happen.

Reply:
That's interesting. I did not know that!